### DISCUSSION TITLE
Make a simple automation based on a attribute under "current_hour"

### DISCUSSION THREAD

**Discussion by rinowit** [erlendsellie/priceanalyzer#8] [Q&A] [answered]
Hello, I would like to make a automation that is just based on "five_cheapest_today" I want the automation to turn a device on when it true and off when it's false. But I can't figure out how I’m supposed to do that when the attribute is under "current_hour" Could you please help me out. 

I love the integration as it gives so many opportunities. Great work

**Comment by rinowit** [comment, marked as answer]
I think I just made it. I was trying to figure ut what a attribute under at attribute is called and googled it. I was sub-attribute. So found a solution.

`"{{ state_attr('sensor.priceanalyzer_oslo', 'current_hour') ['is_five_cheapest'] }}"`

**Comment by erlendsellie** [comment]
Hi! 
Yeah, that looks completely right. 
A 'cleaner look' is to set 'current_hour' as a variable in a template like this, and then check is_five_cheapest, or the attribute you want to use:
```
{%set current_hour = state_attr('sensor.priceanalyzer', 'current_hour') %}
{{current_hour.is_five_cheapest}}

```

**Reply by rinowit** [reply]
Nice, i will try it out. 
I just found a small ting that can confuse people. The attribute is small letters like: true and false. But when you make a sensor from the sub-attribute it say: True and False with big first letter.

**Reply by erlendsellie** [reply]
I'm pretty sure(not a Python expert) that's because Python 'casts' the string 'true' and 'false' to the booelan values True and False (with capitalization) once you create a binary sensor out of it. Attributes are always a string (like a text-value). 

The integration itself sends a boolean (True and False) to the attribute, but it is sent to Home Assistant as strings, and thus becomes true and false with small letters first.

You can see this with other sensors as well, for example occupancy-attributes from Zigbee2mqtt.

Does that make sense?

**Reply by rinowit** [reply]
Absolutely:) 
Just so people are aware of it as if you use wrong first letter the automation won't work.

Your Analyzer is the best price Analyzer out there. I have tried most of what’s out there. And this one is the only one that works and is customizable after you need :)

**Comment by rinowit** [comment]
The integration has worked perfectly the last months. But after updating the HA to the newest version, the template I have been using got broken. 
```
  - platform: template
    sensors:
      five_cheapest:
        friendly_name: Price is 5 cheapest
        value_template: "{{ state_attr('sensor.priceanalyzer_oslo_v1', 'current_hour') ['is_five_cheapest'] }}"
```
I had to stay on priceanalyzer 1.1 to have it to work on the newest HA version. But now after HA may update that doesn’t work either... I just get a blank result. So I had do update to 1.4 and now the sensor just freezes. It stays on just one value (True or false).
You told me to use another template
```
{%set current_hour = state_attr('sensor.priceanalyzer', 'current_hour') %}
{{current_hour.is_five_cheapest}}
```
But i'm not sure how to use that. The last template I had I just added to my configuration.yaml and it was ok. But my HA don’t want to accept this on in my configuration. 
Thanks for a really great integration. It’s just me that is bad on scripting😊

**Comment by erlendsellie** [comment]
Hi @rinowit,
I didn't see your question before now. 
The reason the template i sent you didn't work, was probably since it referenced **sensor.priceanalyzer**, while your priceanalyzer-sensor is called **sensor.priceanalyzer_oslo_v1**. If you just swap out the entity_id, it should work?
Both template's work for me with 2023.9.1 and the latest Priceanalyzer Beta.
